x/evm/types/legacy_tx.go

```diff
@@ -197,7 +197,7 @@ func (tx LegacyTx) Validate() error {
 	if chainID == nil {
 		return errorsmod.Wrap(
 			errortypes.ErrInvalidChainID,
-			"chain ID must be present on AccessList txs",
+			"chain ID must be present on LegacyTx txs",
```

Comment:
this error is wrong. Technically the chain ID is derived on LegacyTxs, not "present"